### PR TITLE
send vendor id with each request

### PR DESCRIPTION
**Pull request by ahmednawar**
@aaustin @derrickstaten cc @scott-branchio

### CODE REVIEW

**Comment by aaustin** [comment]
clean 👍

File: Branch-SDK/Branch-SDK/BNCServerInterface.m [hunk 1]
```diff
@@ -252,6 +252,10 @@ - (void)updateDeviceInfoToMutableDictionary:(NSMutableDictionary *)dict {
         dict[BRANCH_REQUEST_KEY_IS_HARDWARE_ID_REAL] = @(deviceInfo.isRealHardwareId);
     }
     
+    if (deviceInfo.vendorId) {
+        dict[BRANCH_REQUEST_KEY_IOS_VENDOR_ID] = deviceInfo.vendorId;
```
Comment (derrickstaten) [inline Branch-SDK/Branch-SDK/BNCServerInterface.m]: this is exactly what safeSetValue is for. Can you use that instead?

File: Branch-SDK/Branch-SDK/BNCDeviceInfo.m [hunk 2]
```diff
@@ -38,6 +38,11 @@ - (id)init {
             self.hardwareIdType = hardwareIdType;
         }
         
+        NSString *vendorId = [BNCSystemObserver getVendorId:preferenceHelper.isDebug];
+        if (vendorId) {
+            self.vendorId = vendorId;
+        }
+
```
Comment (derrickstaten) [inline Branch-SDK/Branch-SDK/BNCDeviceInfo.m]: it's okay to set to nil, so could we collapse this to one line?

**Comment by ahmednawar** [comment]
@derrickstaten I updated the PR based on your feedback, let me know if you have other comments

**Comment by derrickstaten** [comment]
👍

**Pull request merged**